src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -1201,7 +1062,23 @@ def generate_power_spec_commands(self) -> List[str]:
             self.logger.error(
                 "Invalid power specification type '{tpe}'; only 'cpf' or 'upf' supported".format(tpe=power_spec_type))
             return []
-        power_spec_contents = str(self.get_setting("vlsi.inputs.power_spec_contents"))  # type: str
+
+        power_spec_contents = ""  # type: str
+        power_spec_mode = str(self.get_setting("vlsi.inputs.power_spec_mode"))  # type: str
+        if power_spec_mode == "empty":
+            return []
+        elif power_spec_mode == "auto":
+            if power_spec_type == "cpf":
```

Comment:
I see a world where this is unnecessary, and the correct method is called based on what PNR tool you're using, with overrides if desired. I'll file an issue here too.

Reply:
You can use either cpf or upf for the tool I tested this on. So I thought it best to give the user freedom, if for example one generator produces better output. Perhaps we can have a default per tool for auto.